### ISSUE TITLE
Support for Non default database

### ISSUE THREAD

**Issue by RohanPoojary** [theatlantic/django-nested-admin#167]
# Introduction
Django admin supports  admin panel for custom databases by overriding some core functionalities of `admin` class.
The same needs to be extended to all `NestedInlineModelAdmin`. Thereby enabling users to add nested admin to any configured database.

**Comment by fdintino** [comment]
Have you tried using a non-default database? I don't see a reason why it wouldn't work.

**Comment by RohanPoojary** [comment]
Yes, the reason for this is in `NestedInlineFormSetMixin`. The `queryset` and `delete` functions  uses `default` databases. Thus creating some discrepancies.

I had come across a use case where different admins had to be attached to same model that will be reading on different databases. The approach that I took was to swap database through admin class, instead of modifying model.

I followed below approach for normal Django admin
https://docs.djangoproject.com/en/2.0/topics/db/multi-db/#topics-db-multi-db-hints
This worked.

But similar changes on Nested Admin didn't work. I dwelled through the source code and figured that `NestedInlineFormSetMixin` itself didn't have functionality to operate on non-default database.

**Comment by fdintino** [comment]
Thanks for the detailed reply. I'll look into this.

**Comment by sicorpinc** [comment]
> Have you tried using a non-default database? I don't see a reason why it wouldn't work.

We're using non-default databases due to some performance and security reasons
